openedx/features/course_experience/views/course_dates.py

```diff
@@ -27,10 +29,16 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         course_key = CourseKey.from_string(course_id)
         course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
         course_date_blocks = get_course_date_blocks(course, request.user, request, num_assignments=1)
+        # We will use this boolean to gate if we show a link to the dates tab. This same logic
+        # dictates if we show the tab at all.
+        user_enrolled = (request.user and request.user.is_authenticated and
+                         (bool(CourseEnrollment.is_enrolled(request.user, course.id) or
+                               has_access(request.user, 'staff', course, course.id))))
```

Comment:
I don't think you need the `bool(`, but I guess it's harmless.

Reply:
Yeah, I saw it this way elsewhere and copy-pasted to here. I think we are hopefully close enough to not using this that I'm okay with leaving it.